### PR TITLE
Use travis windows and preseed Github release

### PR DESCRIPTION
**Pull request by adrianbroher**
This is a soon-to-be finished PR, which

* Migrates the build of the Windows SDK to the Travis CI service.
* Removes AppVeyor as Windows CI.
* Preseeds the draft Github release with the changelog and git tag subject.

I would like to integrate this before the new release happens, therefor I would like to ask to put the SDK release on hold, as I may not able to finish this within the next days due to private reasons.

/cc @Vezzra @geoffthemedio @o01eg 

Example release (prepopulated by Travis CI): https://github.com/adrianbroher/freeorion-sdk/releases/tag/test-v10
Current build status (my personal account): https://travis-ci.com/adrianbroher/freeorion-sdk/builds/151226561

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,32 +74,27 @@ jobs:
...
-          tags: true
+
+    - name: Build FreeOrionSDK for Windows
+      <<: *_build_sdk_common
+      os: windows
+      env:
+        - RELEASES_FILE=FreeOrionSDK_*.zip
+        - RELEASES_TAG_NAME=$TRAVIS_TAG
+      script:
+        - cmake -G "Visual Studio 15" -T v141_xp ..
```
Comment (o01eg) [inline .travis.yml]: SDK should be built at least for MSVS 2015 and 2017.
Also I moved CI to use `-A %PLATFORM%` to set platform for cmake.

**Comment by o01eg** [comment]
As I see Travis CI doesn't support multiple MSVS versions, only 2017.

Are you sure we should move to it?

**Comment by geoffthemedio** [comment]
Is MSVC2015 still needed?

**Comment by o01eg** [comment]
I'm fine with dropping MSVC2015 but what about MSVC2019?

**Comment by geoffthemedio** [comment]
Is there info somewhere about whether / when support will be added to Travis CI for MSVC2019?

**Comment by o01eg** [comment]
I don't find any info about different Windows VMs: https://docs.travis-ci.com/user/reference/windows/#supported-languages

**Comment by o01eg** [comment]
Maybe something like this: https://travis-ci.community/t/vs-2019-and-cmake/3796/2

**Comment by adrianbroher** [comment]
There is no reason to care about the generator.  The relevant differences stem from the toolkit and those can be used with with the same generator or a different generator.

**Comment by o01eg** [comment]
@adrianbroher If we can use MSVS 2017 SDK in MSVS 2019 so I'm fine with this PR. Are something other left WIP?

**Comment by adrianbroher** [comment]
Look at the commit list, it's full of shit. isn't that obvious?

**Comment by o01eg** [comment]
Maybe just squash it?

Also what about to don't drop `appveyor.yml` this release? So we got known to work SDK on appveyor and transition SDK on travis.

**Comment by adrianbroher** [comment]
@o01eg how about you sit on your hands for the rest of the day and relax a little?

**Comment by o01eg** [comment]
Another disadvantage of Travis CI is it doesn't have a downloadable artifacts.
With AppVeyor CI it's a very simple to download built SDK to test FreeOrion against it from master branch or even PR to test proposed changes, which was actively used last year.

File: .travis.yml [hunk 2]
```diff
@@ -36,32 +80,37 @@ jobs:
...
-          tags: true
+
+    - name: Build FreeOrionSDK for Windows (x86)
+      <<: *_build_sdk_common
+      os: windows
+      env:
+        - RELEASES_FILE=FreeOrionSDK_*.zip
+        - RELEASES_TAG_NAME=$TRAVIS_TAG
+      script:
+        - cmake -G "Visual Studio 15" -T v141 -A Win32 ..
```
Comment (o01eg) [inline .travis.yml]: Should we drop XP support? @geoffthemedio

Comment (adrianbroher) [reply .travis.yml]: You are trying to hard.  XP EOL is public knowledge.  Also you don't seem to read the  build logs of the stuff you build, do you?  Otherwise you would have seen the deprecation messages regarding the toolkit support (if you don't known what's going on look up MSB8051).  Which is especially strange as you're advocating v142.

TL;DR: You're not in charge of deciding for or against XP support, Microsoft is.

**Comment by adrianbroher** [comment]
> Another disadvantage of Travis CI is it doesn't have a downloadable artifacts.

Which covers the support for 1 out of 3 platforms, which are used for CI right now and is a convenience feature, nothing more.  The target audience for intermediate releases is already present in this thread + @Vezzra and all of them are able to build the SDK from scratch.

However spreading the CI over multiple services makes draft release generation fragile, unreliable and defeats the point of doing automatic releases, where the target audience consists of all current and potential contributors.

@geoffthemedio 
> Does merging this prevent also generating the SDK manually as is currently done, with whatever compiler version is locally available?

No.  Also we don't want 'whatever compiler version is locally available' support, but only the ones, which support for C++11 (or C++14 in future).

**Comment by adrianbroher** [comment]
Removed my PAT from PR, ready for review and merge.

**Comment by geoffthemedio** [comment]
> Also we don't want 'whatever compiler version is locally available' 

Main concern was MSVC2019+ before Travis CI supports it. Keeping semi-up-to-date with supporting newer compiler releases has some advantages beyond better C++ version support.

File: .travis.yml (hunk 2 above)
Comment (geoffthemedio) [reply .travis.yml]: Keeping older versions supported is fine as long as it doesn't impede work severely, but if it does, I think XP is old enough now that we can move on. There are likely some people still using it, but it is almost 20 years old now...

**Comment by o01eg** [comment]
@geoffthemedio Looks like I got something it work: https://github.com/freeorion/freeorion-sdk/pull/64/commits/52540eb654797e9504b467cffef610bca83cda79

At least cmake found MSVS2019 toolset.

**Comment by o01eg** [comment]
It's even broken same way it is broken on AppVeyor )

**Comment by adrianbroher** [comment]
> Main concern was MSVC2019+ before Travis CI supports it. Keeping semi-up-to-date with supporting newer compiler releases has some advantages beyond better C++ version support.

And this is releated to the current status of the SDK how?

The main project uses the v14.0 XP and v14.1 XP toolkit (which is already one to many in terms of maintainance). Unless you change this there is no need to provide any variant of the SDK.  Same goes for the CPU arch. @o01eg persistence on this topic is not understandable.  It wastes developer effort and CI time for something that is not used.

Also please distinguish between IDE and toolkit, those concepts are unrelated.  You can use Visual Studio version whatever with pretty much any toolkit version.  The only relevant part for the project  is the toolkit, which bundles the compiler feature set and the platform SDK.

**Comment by Vezzra** [comment]
Tyring to reduce the number of different services used makes sense to me. Building the SDK isn't that hard, so loosing the convenience of having downloadable build artifacts shouldn't be a concern IMO. So that shouldn't keep this PR from getting merged.

As @adrianbroher has declared the PR ready for review and merge, I also don't see a reason not to wait with the v11 release of the SDK until this gets merged, assuming that can be done within a few days at most.

@o01eg, @geoffthemedio, are there any concerns aside from Travis not providing downloadable artifacts? Are the proposed changes ok?

**Comment by o01eg** [comment]
@Vezzra Concern about MSVS2019+ is solvable with installation new MSVS version. I suppose there no other concerns left.

**Comment by Vezzra** [comment]
Excellent, now we only need feedback from @geoffthemedio, and we should be good to go 😃

**Comment by geoffthemedio** [comment]
Didn't realize I needed to comment on this.

Given the relative rarity of SDK updates, I don't think it's a problem if they have to be uploaded manually.

**Comment by o01eg** [comment]
When it will be released?

**Comment by adrianbroher** [comment]
When the Changelog has the correct date and an annotated tag was created.

The process is not automatic and never will be.

**Comment by o01eg** [comment]
@adrianbroher Who are going to do it?

**Comment by Vezzra** [comment]
Assuming the SDK release procedure @adrianbroher explained to me two years ago hasn't changed (@adrianbroher, correct me if that assumption no longer applies), I'm going to do that.

Just need to make sure the changelog is up-to-date, see also the [thread concerning the SDK release](https://freeorion.org/forum/viewtopic.php?f=9&t=11562) I opened on our forum.
